addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -130,6 +134,16 @@ def _record_production(self):
                 'date_start': datetime.now(),
             })
 
+    def _check_strict_consumption(self, lines_to_check):
+        lines_to_check = super(MrpProductProduce, self)._check_strict_consumption(lines_to_check)
+        for line in lines_to_check:
+            same_pp_line = self.workorder_line_ids.filtered(lambda l: l.product_id == line.product_id)
+            qty_done = sum(same_pp_line.mapped('qty_done'))
+            qty_to_consume = sum(same_pp_line.mapped('qty_to_consume'))
+            rounding = self.product_uom_id.rounding
+            if float_compare(qty_done, qty_to_consume, precision_rounding=rounding) == -1:
```

Comment:
duplicated block, use a kwarg in the original method on abstract workorder and change it when used from the workorders